### ISSUE TITLE
No Binary Weights

### ISSUE THREAD

**Issue by lizhiyuanUSTC** [jiecaoyu/XNOR-Net-PyTorch#84]
Hi, I read your code carefully, but I didn't find that the weight was binarized.
`

    class BinConv2d(nn.Module): # change the name of BinConv2d
        def __init__(self, input_channels, output_channels,
                kernel_size=-1, stride=-1, padding=-1, groups=1, dropout=0,
                Linear=False):
            super(BinConv2d, self).__init__()
            self.layer_type = 'BinConv2d'
            self.kernel_size = kernel_size
            self.stride = stride
            self.padding = padding
            self.dropout_ratio = dropout

            if dropout!=0:
                self.dropout = nn.Dropout(dropout)
            self.Linear = Linear
            if not self.Linear:
                self.bn = nn.BatchNorm2d(input_channels, eps=1e-4, momentum=0.1, affine=True)
                self.conv = nn.Conv2d(input_channels, output_channels,
                        kernel_size=kernel_size, stride=stride, padding=padding, groups=groups)
            else:
                self.bn = nn.BatchNorm1d(input_channels, eps=1e-4, momentum=0.1, affine=True)
                self.linear = nn.Linear(input_channels, output_channels)
            self.relu = nn.ReLU(inplace=True)
    
        def forward(self, x):
            x = self.bn(x)
            x = BinActive()(x)
            if self.dropout_ratio!=0:
                x = self.dropout(x)
            if not self.Linear:
                x = self.conv(x)
            else:
                x = self.linear(x)
            x = self.relu(x)
            return x

`

**Comment by jiecaoyu** [comment]
@lizhiyuanUSTC hi, weights are binarized [here](https://github.com/jiecaoyu/XNOR-Net-PyTorch/blob/ebccb503b2f9ebc96187a75b6daa63d95b61d110/ImageNet/networks/main.py#L244).
